### ISSUE TITLE
Connect to Docker endpoint with self-signed cert

### ISSUE THREAD

**Issue by ipmb** [mozilla-services/heka#1279]
There's no way (I can tell) to connect to a Docker endpoint using SSL with a self-signed certificate.

Config:

```
[DockerLogInput]
endpoint = "https://192.168.59.103:2376"
```

Output:

```
2015/01/19 18:30:11 Loading: [DockerLogInput]
2015/01/19 18:30:11 Initialization failed for 'DockerLogInput': DockerLogInput: failed to attach: Get https://192.168.59.103:2376/containers/json: x509: certificate signed by unknown authority
2015/01/19 18:30:11 Loading: [DashboardOutput]
2015/01/19 18:30:11 Error reading config: 1 errors loading plugins
```

It'd be nice to be able to specify a cert path or just disable TLS checking (for local testing). The use case is [boot2docker](http://boot2docker.io/) which does something like this to connect the client:

```
$ boot2docker shellinit
Writing /Users/pete/.boot2docker/certs/boot2docker-vm/ca.pem
Writing /Users/pete/.boot2docker/certs/boot2docker-vm/cert.pem
Writing /Users/pete/.boot2docker/certs/boot2docker-vm/key.pem
    export DOCKER_HOST=tcp://192.168.59.103:2376
    export DOCKER_CERT_PATH=/Users/pete/.boot2docker/certs/boot2docker-vm
    export DOCKER_TLS_VERIFY=1
```

**Comment by carlanton** [comment]
I think this could be pretty easy to implement. fsouza/go-dockerclient implemented a TLS support in december. I'll give it a try this weekend!

**Comment by carlanton** [comment]
This problem is now solved in Heka 0.9 with the new config parameter ```cert_path```:
https://hekad.readthedocs.org/en/v0.9.0/config/inputs/docker_log.html
